kolibri/plugins/learn/assets/src/views/LearnIndex.vue

```diff
@@ -387,7 +387,8 @@
       },
       handleSubmitUpdateYourProfileModal() {
         if (this.userProfilePluginUrl) {
-          const url = `${this.userProfilePluginUrl()}#/profile/edit?next_page=learn`;
+          const next = window.encodeURIComponent(window.location.href);
+          const url = `${this.userProfilePluginUrl()}#/profile/edit?next=${next}`;
```

Comment:
The modal which triggers this handler is conditional on `v-if="profileNeedsUpdate"`

```vue
    <UpdateYourProfileModal
      v-if="profileNeedsUpdate"
```

If the profile may be turned off completely, the modal should never be displayed and this handler should never be called. Therefore we should probably update the entire modal to be conditional on that:

```vue
    <UpdateYourProfileModal
      v-if="profileNeedsUpdate && userProfilePluginUrl"
```

and remove the condition from the 'submit' handler:

```js
      handleSubmitUpdateYourProfileModal() {
        const next = window.encodeURIComponent(window.location.href);
        const url = `${this.userProfilePluginUrl()}#/profile/edit?next=${next}`;
        const redirect = () => {
          window.location.href = url;
        };
        this.$store
          .dispatch('deferProfileUpdates', this.demographicInfo)
          .then(redirect, redirect);
      },
```